test/integration/scheduler/preemption_test.go

```diff
@@ -355,4 +357,220 @@ var _ = ginkgo.Describe("Preemption", func() {
 		})
 	})
 
+	ginkgo.Context("When most quota is in a shared ClusterQueue in a cohort", func() {
```

Comment:
Maybe we can mix a few cases into a single more complex scenario:
for example:
There are running workloads in standard A and B and best effort, playing with different priorities.
A new workload comes in in standard A, only best effort workloads get preempted.

Reply:
Yeah, I was hesitant if multiple small scenarios or one bigger is better. I have now updated to have one bigger. PTAL.

Reply:
We can leave small scenarios for unit tests.

Reply:
Sure, we have unit test scenario corresponding to the removed integration test scenarios in `pkg/scheduler/preemption/preemption_test.go`. I have reviewed the unit test descriptions in the recently pushed commit.